frontend/src/pages/CompareV2.tsx

```diff
@@ -292,9 +313,74 @@ function getDropdownItems(filteredRunArtifacts: RunArtifact[]) {
   return dropdownItems;
 }
 
+function getArtifactFromSelectedItem(
+  filteredRunArtifacts: RunArtifact[],
+  selectedItem: SelectedItem,
+): Artifact | undefined {
+  const filteredRunArtifact = filteredRunArtifacts.find(
+    runArtifact => runArtifact.run.run?.name === selectedItem.itemName,
+  );
+
+  const executionArtifact = filteredRunArtifact?.executionArtifacts.find(executionArtifact => {
+    const executionText: string =
+      getExecutionName(executionArtifact.execution) ||
+      executionArtifact.execution.getId().toString();
+    return executionText === selectedItem.subItemName;
+  });
+
+  const artifact = executionArtifact?.linkedArtifacts.find(linkedArtifact => {
+    const linkedArtifactText: string =
+      getArtifactName(linkedArtifact) || linkedArtifact.artifact.getId().toString();
+    return linkedArtifactText === selectedItem.subItemSecondaryName;
+  })?.artifact;
+
+  return artifact;
+}
+
 function MetricsDropdown(props: MetricsDropdownProps) {
-  const { filteredRunArtifacts, metricsTabText } = props;
-  const [selectedItem, setSelectedItem] = useState<SelectedItem>({ itemName: '', subItemName: '' });
+  const {
+    filteredRunArtifacts,
+    metricsTab,
+    metricsTabText,
+    selectedArtifacts,
+    setSelectedArtifacts,
+  } = props;
+  const firstSelectedArtifact: SelectedArtifact = selectedArtifacts[metricsTab][0];
+  const secondSelectedArtifact: SelectedArtifact = selectedArtifacts[metricsTab][1];
+
+  const [firstSelectedItem, setFirstSelectedItem] = useState<SelectedItem>(
+    firstSelectedArtifact.selectedItem,
+  );
+  const [secondSelectedItem, setSecondSelectedItem] = useState<SelectedItem>(
+    secondSelectedArtifact.selectedItem,
+  );
+
+  const updateSelectedArtifacts = useCallback(
+    (selectedItem: SelectedItem, panelIndex: number): void => {
+      if (
+        selectedItem.itemName &&
+        selectedItem.subItemName &&
+        selectedItem.subItemSecondaryName &&
+        selectedArtifacts[metricsTab][panelIndex].selectedItem !== selectedItem
+      ) {
+        selectedArtifacts[metricsTab][panelIndex].selectedItem = selectedItem;
+        selectedArtifacts[metricsTab][panelIndex].selectedArtifact = getArtifactFromSelectedItem(
+          filteredRunArtifacts,
+          selectedItem,
+        );
+        setSelectedArtifacts({ ...selectedArtifacts });
+      }
+    },
+    [filteredRunArtifacts, metricsTab, selectedArtifacts, setSelectedArtifacts],
+  );
+
+  useEffect(() => {
+    updateSelectedArtifacts(firstSelectedItem, 0);
+  }, [firstSelectedItem, updateSelectedArtifacts]);
+
+  useEffect(() => {
+    updateSelectedArtifacts(secondSelectedItem, 1);
+  }, [secondSelectedItem, updateSelectedArtifacts]);
```

Comment:
Can we simplify/omit this logic by embedding it to the `setSelectedItem` callback in `<TwoLevelDropdown>` component? See my other comment in `<TwoLevelDropdown>`.

Reply:
Yes we can - I referenced the commit on the other discussion: https://github.com/kubeflow/pipelines/pull/7966#discussion_r915093455.